app/content/profile/index.js

```diff
@@ -103,9 +102,30 @@ function Index(props) {
   }, [userData, userEmail]);
 
   useEffect(() => {
-    setSnapPoints(
-      passwordInputVisible ? expandedSnapPoints : initialSnapPoints,
+    setSnapPoints([passwordInputVisible ? "70%" : "57%"]);
+  }, [passwordInputVisible]);
+
+  useEffect(() => {
+    const keyboardDidShowListener = Keyboard.addListener(
```

Comment:
I'll have to look at this more closely later but I think you can simplify a lot of this by using this library (we use it for keyboard inputs elsewhere, e.g. signin)

https://www.npmjs.com/package/react-native-keyboard-aware-scroll-view